src/Orleans/Facet/FacetAttribute.cs

```diff
@@ -0,0 +1,12 @@
+﻿
+using System;
+
+namespace Orleans
+{
+    /// <summary>
+    /// Marker attribute for facets
+    /// </summary>
+    public abstract class FacetAttribute : Attribute
```

Comment:
>If we went with an interface instead, we could more easily change the requirement later to open it up to any attribute type

Does it make it more easy because we'd have to write attribute filtering logic now, so it would be easier to modify that logic later?
I don't see much difference between using an interface, base class, or base attribute marker, other than using GetCustomAttributes() + filtering the attributes ourselves instead of using GetCustomAttribute<T>().
I'm not opposed to some other marker, I'm just not seeing how it leads to "more easily change the requirement later", and it doesn't enable the scenarios Julian seems interested in...